framework/src/modules/interoperability/utils.ts

```diff
@@ -40,3 +40,27 @@ export const getCCMSize = (ccm: CCMsg) => {
 
 	return serializedCCM.byteLength;
 };
+
+export const getEncodedSidechainTerminatedCCMParam = (
+	ccm: CCMsg,
+	receivingChainAccount: ChainAccount,
+) => {
+	const params = {
+		chainID: ccm.receivingChainID,
+		stateRoot: receivingChainAccount.lastCertificate.stateRoot,
+	};
+
+	const encodedParams = codec.encode(sidechainTerminatedCCMParamsSchema, params);
+
+	return encodedParams;
+};
+
+export const nullOnError = async <T>(promise: Promise<T>) => {
+	let result;
+	try {
+		result = await promise;
+	} catch {
+		result = null;
+	}
+	return result;
+};
```

Comment:
maybe? 
```suggestion
export const handlePromise = async <T>(promise: Promise<T>): Promise<T | undefined> => {
	let result;
	try {
		result = await promise;
		return result;
	} catch {
		return;
	}
};
```

Reply:
Result is defined on existence of a value, so I think it must be defined in absence of a value to denote a processing happened. In this case result is null and not undefined because it is defined and it's absent.